addOns/client/src/main/java/org/zaproxy/addon/client/spider/ClientSpiderDialog.java

```diff
@@ -305,7 +366,11 @@ public void save() {
 
         subtreeOnlyPreviousCheckedState = getBoolValue(FIELD_SUBTREE_ONLY);
 
-        this.extension.runSpider(getStartUrl(), params);
+        User user = this.getSelectedUser();
+        this.extension
+                .getAuthenticationHandlers()
+                .forEach(handler -> handler.enableAuthentication(user));
```

Comment:
This should be done by/in the spider, otherwise we'll have to duplicate this code throughout (e.g. API, automation, …), also better to have the code that enables/disables closer.